#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_fix-asserts.dpatch by Henrique de Moraes Holschuh <hmh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: get rid of code that causes warnings under NDEBUG builds

@DPATCH@
diff -urNad hplip~/prnt/hpijs/colormatch.cpp hplip/prnt/hpijs/colormatch.cpp
--- hplip~/prnt/hpijs/colormatch.cpp	2006-05-10 20:22:01.000000000 -0300
+++ hplip/prnt/hpijs/colormatch.cpp	2006-05-10 20:30:52.594599906 -0300
@@ -221,15 +221,21 @@
 {
     for (unsigned int r = 0; r < 9; r++)
     {
+#ifndef NDEBUG
         unsigned long ul_RedMapPtr = r * 9 * 9;
+#endif
         for (unsigned int g = 0; g < 9; g++)
         {
+#ifndef NDEBUG
             unsigned long ul_GreenMapPtr = g * 9;
+#endif
             for (unsigned int b = 0; b < 9; b++)
             {
                 unsigned long mapptr = b + (g * 9) + (r * 9 * 9);       // get address in map
+#ifndef NDEBUG
                 unsigned long ul_MapPtr = b + ul_GreenMapPtr + ul_RedMapPtr;
                 ASSERT(mapptr == ul_MapPtr);
+#endif
                 // put r,g,b in monitor range
                 unsigned int oldR = r * 255 >> 3;
                 unsigned int oldG = g * 255 >> 3;
